### PR TITLE
fix: isolated CallOptions by Plugin

### PR DESCRIPTION
**Pull request by falconandy**

### CODE REVIEW

File: packages/sdk/lib/base.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export type Properties = {
   [name: string]: any;
 };
 
-export type CallOptions = Record<string, any>;
+export type CallOptions = Record<string, unknown>;
```
Comment (justin-iteratively) [inline packages/sdk/lib/base.ts:9]: @falconandy Why `unknown` vs `any`?

File: packages/sdk/lib/base.ts [hunk 2]
```diff
@@ -99,7 +99,7 @@ export abstract class Plugin {
     validationResponses: ValidationResponse[],
   ): void {}
 
-  track(userId: string | undefined, event: Event, options?: TrackOptions): void {}
+  track(userId: string | undefined, event: Event, options?: unknown): void {}
```
Comment (justin-iteratively) [inline packages/sdk/lib/base.ts:102]: Is `unknown` here necessary to allow the Plugin implementations to type cast to their own <Plugin>TrackOptions?

File: packages/plugin-amplitude-node/lib/index.ts [hunk 3]
```diff
@@ -51,8 +49,8 @@ export class AmplitudePlugin extends RequestLoggerPlugin {
     this.amplitude = new Amplitude(this.apiKey, this.options);
   }
 
-  async identify(userId: string, properties?: Properties, options?: IdentifyOptions) {
-    const { callback } = this.getPluginCallOptions<AmplitudeIdentifyOptions>(options);
+  async identify(userId: string, properties?: Properties, options?: AmplitudeIdentifyOptions) {
```
Comment (justin-iteratively) [inline packages/plugin-amplitude-node/lib/index.ts:52]: The typecast here is awesome. I was hoping we could do that but wasn't sure if we could.

File: packages/sdk/lib/base.ts (hunk 1 above)
Comment (falconandy) [reply packages/sdk/lib/base.ts:9]: `unknown` is stricter

File: packages/sdk/lib/base.ts (hunk 2 above)
Comment (falconandy) [reply packages/sdk/lib/base.ts:102]: `any` should work too

**Review by justin-iteratively — APPROVED** [review]

**Pull request merged**